### PR TITLE
Add Python bindings.

### PR DESCRIPTION
**Pull request by chr1sj0nes**
#954

### CODE REVIEW

**Comment by LebedevRI** [comment]
This is missing licence header blobs and cmake stuff.

**Comment by dominichamon** [comment]
Great! I have two requests:

1. Can we turn the example into a py_test? I know it doesn't actually test anything, but if we can then run it on travis we get some level of protection against someone introducing a bug that affects the bindings.

2. is there any documentation we can add (to https://github.com/google/benchmark/tree/master/docs, linked from the main README.md)?

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    version=_get_version(),
+    url='https://github.com/google/benchmark',
+    description='A library to benchmark code snippets.',
+    author='Google',
+    author_email='benchmark-py@google.com',
+    # Contained modules and scripts.
+    package_dir={'': 'bindings/python'},
+    packages=setuptools.find_packages('bindings/python'),
+    install_requires=_parse_requirements('bindings/python/requirements.txt'),
+    cmdclass=dict(build_ext=BuildBazelExtension),
```
Comment (LebedevRI) [inline setup.py:108]: Hmm.
Reading up on this, and i think either the cmake build system needs to be explicitly deprecated,
or this should be cmake-based.

Comment (dominichamon) [reply setup.py:108]: We could have the Python bindings requiring bazel (given they require absl which doesn't have a good cmake solution). No?

**Comment by chr1sj0nes** [comment]
> This is missing licence header blobs and cmake stuff.

Ah... I checked and saw that many files didn't have the copyright / license header, but I've just seen that the public headers do. Which files require it? The published files (the Python in this case), or all of them?

What do you mean by CMake stuff? Is is not okay to rely on Bazel for building the Python extension?

**Comment by chr1sj0nes** [comment]
> This is missing licence header blobs and cmake stuff.

I've added the license headers to the two Python files that will be distributed. Are they needed anywhere else?

**Comment by dominichamon** [comment]
If someone wants the python bindings,  they need to use blaze, right? i think that's ok for now and if someone needs cmake we/they can always add that.

**Comment by chr1sj0nes** [comment]
> If someone wants the python bindings, they need to use bazel, right? i think that's ok for now and if someone needs cmake we/they can always add that.

Yes, that's correct

**Comment by LebedevRI** [comment]
On Wed, Apr 15, 2020 at 2:30 PM Chris Jones <notifications@github.com>
wrote:

> This is missing licence header blobs and cmake stuff.
>
> Ah... I checked and saw that many files didn't have the copyright /
> license header, but I've just seen that the public headers do. Which files
> require it? The published files (the Python in this case), or all of them?
>
Every file, IMO

> What do you mean by CMake stuff? Is is not okay to rely on Bazel for
> building the Python extension?
>
I guess i'll just refrain from the review then :)

**Comment by chr1sj0nes** [comment]
Hi Dominic. Apologies for my procrastination on this. I've changed the example to be a `py_test`. Is there anything else that needs fixing before this can go in?

**Comment by dominichamon** [comment]
i think it's great. maybe a markdown doc linked from the main README to complete the documentation story would be ideal if you can, but if you want to do that as a followup let me know and i'll land this.

**Comment by chr1sj0nes** [comment]
It would be great to get this in. If I can do the documentation in a follow-up, that would be preferable to me.

**Pull request merged**